### PR TITLE
BAU: Improve Exceptions + Retry RP stub steps

### PR DESCRIPTION
**Pull request by whi-tw**
## What

Details all in individual commits.

Could potentially be referred to as a kitchen sink, sorry!

## How to review

1. Ensure you can run tests locally (ie. have a valid `.env` file) 
2. Run the tests with `./run-acceptance-tests.sh -l build`
3. Observe that all the tests pass

*OverAchiever Steps*:

- You could potentially modify a test to definitely fail, and see if the exceptions work properly.
- You could modify the 'wait for' test in the modified rp stub initialisation to trigger a fail and see if the retries work properly.

### CODE REVIEW

File: acceptance-tests/src/test/java/uk/gov/di/test/utils/RetryHelper.java [hunk 1]
```diff
@@ -0,0 +1,31 @@
+package uk.gov.di.test.utils;
+
+import java.util.HashMap;
+import java.util.concurrent.atomic.AtomicInteger;
+
+public class RetryHelper {
+
+    private static volatile RetryHelper instance;
```
Comment (dbes-gds) [inline acceptance-tests/src/test/java/uk/gov/di/test/utils/RetryHelper.java:8]: think volatile is for system or low level development

Comment (whi-tw) [reply acceptance-tests/src/test/java/uk/gov/di/test/utils/RetryHelper.java:8]: > Every thread that accesses a volatile variable will read it from main memory, and not from the CPU cache. This way, all threads see the same value for the volatile variable.

Comment (dbes-gds) [reply acceptance-tests/src/test/java/uk/gov/di/test/utils/RetryHelper.java:8]: ok nice, genuinely have never seen anyone use this :)

**Review by dbes-gds — DISMISSED** [review]

File: acceptance-tests/src/test/java/uk/gov/di/test/pages/BasePage.java [hunk 2]
```diff
@@ -1,24 +1,18 @@
...
 
 import io.cucumber.java.Scenario;
-import org.openqa.selenium.By;
-import org.openqa.selenium.Cookie;
-import org.openqa.selenium.JavascriptExecutor;
-import org.openqa.selenium.OutputType;
-import org.openqa.selenium.TakesScreenshot;
-import org.openqa.selenium.WebDriver;
-import org.openqa.selenium.WebElement;
+import org.openqa.selenium.*;
```
Comment (BeckaL) [inline acceptance-tests/src/test/java/uk/gov/di/test/pages/BasePage.java:4]: Think your editor has probably auto wildcard imported here, we try and avoid this as per https://gds-way.digital.cabinet-office.gov.uk/manuals/programming-languages/java.html#imports

**Review by BeckaL — COMMENTED** [review]
Approve from a general code perspective (subject to fixing comment about wildcard imports) . I think it might be good to get @Gosifo to do a sense check and also test it out

File: acceptance-tests/src/test/java/uk/gov/di/test/pages/BasePage.java (hunk 2 above)
Comment (whi-tw) [reply acceptance-tests/src/test/java/uk/gov/di/test/pages/BasePage.java:4]: roger dodger, i'll sort it!

**Review by dbes-gds — APPROVED** [review]

**Pull request merged**